### PR TITLE
Update deploy script to support automated deploys to production

### PR DESCRIPTION
**Pull request by gaganawhad**

### CODE REVIEW

**Comment by gaganawhad** [comment]
@benhutton @joeosburn-desiringgod @pjmelling  Here's the deploy guide I am proposing for automated deploys to production. Can you take a look at it and give me feedback or a sign-off? 

After we are good with this, I'll set up the missing pieces to make this happen.

**Comment by joeosburn-desiringgod** [comment]
@gaganawhad Adding some clarification for points 5, 6, and 7 on how someone knows if the steps are needed would be helpful.

**Comment by joeosburn-desiringgod** [comment]
@gaganawhad What happens if the master CI build fails (most probably due to a spurious failure)?

**Comment by gaganawhad** [comment]
@joeosburn-desiringgod yeah, good point about points 5, 6 and 7. Any suggestions on how to update?

**Comment by gaganawhad** [comment]
@joeosburn-desiringgod Regarding spurious or other failures on CI builds I have added this: 
> (Fix the CI build if it does not pass)

**Comment by joeosburn-desiringgod** [comment]
@gaganawhad Regarding the spurious failure, if you fix, will it automatically deploy once it's fixed? A little clarification on what would happen. And how to fix (do you just re-run in circle CI?) I think would be helpful.

As far as points 5, 6, and 7, perhaps some language about how you should know if you need to based on the needs of your branch, but the default is that you will not need to.

Thanks!!

File: workflow/deploying/README.md [hunk 1]
```diff
@@ -3,92 +3,55 @@
...
 
-### Documentation
+1. Making sure you have enough room for supporting the release (give yourself a few hours after the deploy), **merge
+  your branch** into remote master.
 
-1. [Create tag and release notes on github](https://github.com/desiringgod/desiring-god-new/releases), following the established pattern (Pull Requests, Pre-deploy actions, Post-deploy actions), pasting in PR titles and numbers as patterned.
+  This will:
 
-### Deployment
+  - Trigger a CI build. (Fix the CI build if it does not pass)
```
Comment (benhutton) [inline workflow/deploying/README.md]: @gaganawhad "Fix" should probably be changed to "Rebuild"

**Comment by gaganawhad** [comment]
> As far as points 5, 6, and 7, perhaps some language about how you should know if you need to based on the needs of your branch, but the default is that you will not need to.

@joeosburn-desiringgod That's where I am hoping I could get some suggestions! :)

File: workflow/deploying/README.md (hunk 1 above)
Comment (gaganawhad) [inline workflow/deploying/README.md]: 👍

**Comment by joeosburn-desiringgod** [comment]
@gaganawhad From @benhutton:

```

benhutton [1:25 PM]  
what if we just said something like, "Immediately upon deploy, view your changes in production. If something doesn't look right, alert the rest of the team and we'll work with you to figure out a fix."
```

**Comment by benhutton** [comment]
(And then you can say something like, "Possible fixes include clearing the cache, restarting heroku, running a script you forgot, or even writing and deploying new code.)

**Comment by gaganawhad** [comment]
@benhutton @joeosburn-desiringgod @pjmelling done!

File: workflow/deploying/README.md [hunk 2]
```diff
@@ -3,102 +3,72 @@
...
+  - In-case there is a problem with the release, you can always rollback to the previous version.
+    Use:
 
-### Pre-Deploy QA
+    ```
+    heroku rollback
+    ```
 
-1. Make sure master build is passing.
+    More information [here](https://devcenter.heroku.com/articles/releases#rollback)
```
Comment (benhutton) [inline workflow/deploying/README.md]: @gaganawhad this should end with a period.

**Comment by gaganawhad** [comment]
@benhutton I fixed that and added a couple other changes.

File: workflow/deploying/README.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ How to deploy to desiringgod.org's production server.
     heroku rollback
     ```
 
-    More information [here](https://devcenter.heroku.com/articles/releases#rollback).
+    More information [here](https://devcenter.heroku.com/articles/releases#rollback)
```
Comment (benhutton) [inline workflow/deploying/README.md]: @gaganawhad you lost your period here again!

**Comment by gaganawhad** [comment]
@benhutton take another look.

**Comment by benhutton** [comment]
:+1: let's get this squashed and merged!

**Pull request merged**